test/rules/do_not_create_setter_without_getter.dart

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// test w/ `pub run test -N do_not_create_setter_without_getter`
+
+class A {
+  // ignore: unused_field
+  int _x;
+
+  set x(int x) { // LINT
+    this._x = x;
+  }
+}
+
+class B extends A {
+  @override
+  set x(int x) { // OK because it is an inherited setter.
+    this._x = x;
+  }
+}
+
+class C {
+  int _x;
+
+  int get x => _x;
+}
+
+class D extends C {
+  set x(int x) { // OK because has inherited getter.
+    this._x = x;
+  }
+}
```

Comment:
You should add a test that there is no lint if the getter and setter are in the same class.